protobufs/livekit_models.proto

```diff
@@ -131,6 +131,7 @@ message ParticipantInfo {
   bool is_publisher = 13;
   Kind kind = 14;
   map<string, string> attributes = 15;
+  DisconnectReason disconnect_reason = 16;
```

Comment:
Sure. If that is `Participant` specific, I think it should be okay. Thoughts on that @davidzhao .

Currently, internally server has these reasons for participant close - https://github.com/livekit/livekit/blob/5d554867102d8b3bee6e8865a85cae5513c19c62/pkg/rtc/types/interfaces.go#L84 and it gets mapped to `DisconnectReason` here - https://github.com/livekit/livekit/blob/5d554867102d8b3bee6e8865a85cae5513c19c62/pkg/rtc/types/interfaces.go#L169.

We can make more participant close reasons and map them to reasons like you mention above.

We do have participant kind now. Maybe, each kind has more specific reasons.

Question: I do not know enough about how SIP participant works. How would the SIP participant be conveyed the reason for its close? Actually, I guess I should first learn how the SIP participant gets closed now.

Reply:
Currently it just disconnects from the room as usual (or is remove from the room by API). So I'll need to update the SDK to expose the disconnect reason (or at least some of them).

Reply:
Got it. Client leaving currently sends this https://github.com/livekit/protocol/blob/a77139eae2660bc3d88e4440a174d8287b07bcf9/protobufs/livekit_rtc.proto#L39 and server just marks it as client initiated leave. If clients are going to send a reason, we can update server to parse the reason and mark the proper reason (would be mapping from `DisconnectReason` -> `ParticipantCloseReason` and back to `DisconnectReason`, but that is okay 🙂 )

Reply:
I think we can add these as new disconnect reasons for SIP.

Reply:
Thank you @davidzhao .

@dennwc Please let me know when you add these to `LeaveRequest`. We can then change the server to do the mapping.  Maybe call the reasons with a `SIP_` prefix for `UNAVAILABLE` and `REJECTED`. I do not particularly like prefixing with specific like that, but the un-prefixed version is very generic. So, that could be confusing.

Reply:
yeah hangup makes sense.. but for the rest, how about keeping it the original SIP reasons instead of mapping? this way we wouldn't need to prefix either

No response -> NO_RESPONSE
Busy -> USER_BUSY (not sure about the USER_, just wanted to disambiguate from server busy)